external/fv3fit/fv3fit/data/tfdataset.py

```diff
@@ -32,9 +52,8 @@ def get_record(self, name: str, ds: xr.Dataset, unstacked_dims: Sequence[str]):
                 raise ValueError("variable {} has no dimension {}".format(name, dim))
         if self.times == "start":
             ds = ds.isel(time=0)
-        dims = [d for d in unstacked_dims if d in ds[name].dims]
-        data = ds[name].transpose(*dims).values
-        return data
+        ds = stack(ds, unstacked_dims)
```

Comment:
Took me a long time to realize that the prior functionality didn't do any stacking at all, and that this PR adds it.

Reply:
Yeah, I should have said that more explicitly. Sorry!